src/platforms/wayland/displayclient.cpp

```diff
@@ -294,17 +308,22 @@ void mgw::DisplayClient::Output::done(void* data, struct wl_output* /*wl_output*
 
 void mgw::DisplayClient::Output::for_each_display_buffer(std::function<void(DisplayBuffer & )> const& f)
 {
-    if (!window)
+    if (!shell_surface)
     {
-        static wl_shell_surface_listener const shell_surface_listener{
-            [](void*, wl_shell_surface* shell_surface, uint32_t serial){ wl_shell_surface_pong(shell_surface, serial); },
-            [](void*, wl_shell_surface*, uint32_t, int32_t, int32_t){},
-            [](void*, wl_shell_surface*){}
+        static xdg_surface_listener const shell_surface_listener{
+            [](void*, auto xdg_surface, auto serial) { xdg_surface_ack_configure(xdg_surface, serial); },
```

Comment:
The configure is acked here. The protocol is worded a bit weird, but the `.ack_configure` should be in response to `xdg_surface.configure` (which commits the new state), not the preceding `xdg_toplevel.configure` (which only sets the pending state).